app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationDebugTest.kt

```diff
@@ -0,0 +1,399 @@
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.SyncStatusModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.networking.NetworkConnectionDebugUtilModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.oppia.android.util.threading.BackgroundDispatcher
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import org.robolectric.shadows.ShadowToast
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests to verify the working of Platform Parameter Architecture. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterIntegrationDebugTest.TestApplication::class)
+class PlatformParameterIntegrationDebugTest {
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+
+  @Inject lateinit var context: Context
+  @Inject lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+  @Inject lateinit var platformParameterSyncUpWorkerFactory: PlatformParameterSyncUpWorkerFactory
+  @Inject lateinit var platformParameterController: PlatformParameterController
+
+  private companion object {
+    private const val REMOTE_DATABASE_NAME = "platform_parameter_and_feature_flag_database"
+    private const val SPLASH_MESSAGE = "Welcome User"
+  }
+
+  fun setUp() {
+    setUpTestApplicationComponent()
+    val config = Configuration.Builder()
+      .setExecutor(SynchronousExecutor())
+      .setWorkerFactory(platformParameterSyncUpWorkerFactory)
+      .build()
+    platformParameterController.loadParametersAsync()
+    testCoroutineDispatchers.runCurrent()
+    WorkManagerTestInitHelper.initializeTestWorkManager(context, config)
+    TestActivity.registerWithPackageManager<SplashTestActivity>(context)
+  }
+
+  @Test
+  fun testIntegration_readEmptyDatabase_checkWelcomeMsgIsInvisibleByDefault() {
+    setUp()
+    launch(SplashTestActivity::class.java).use { scenario ->
+      // Fetch the latest platform parameter from cache store after execution of work request to
+      // imitate the loading process at the start of splash test activity.
+      scenario.onActivity { activity ->
+        activity.splashTestActivityPresenter.loadPlatformParameters()
+      }
+      testCoroutineDispatchers.runCurrent()
+
+      assertThat(ShadowToast.getLatestToast()).isNull()
+    }
+  }
+
+  @Test
+  fun testIntegration_readRemoteDatabase_checkWelcomeMsgIsVisible() {
```

Comment:
Preferring as it states the intent of precedence -- but will also align with any changes on `codeowner` reviews later.
```suggestion
  fun testIntegration_withRemoteNoLocalOverride_remoteTakesPrecedence_displaysWelcomeMsg() {
```

Reply:
updated